packages/ckb-indexer/src/indexer.ts

```diff
@@ -0,0 +1,250 @@
+    return new Promise((resolve) => setTimeout(resolve, timeout));
+  }
+
+  async waitForSync(blockDifference = 0): Promise<void> {
+    const rpcTipNumber = parseInt(
+      (await this.getCkbRpc().get_tip_header()).number,
+      16
+    );
+    while (true) {
+      const indexerTipNumber = parseInt((await this.tip()).block_number, 16);
+      if (indexerTipNumber + blockDifference >= rpcTipNumber) {
+        return;
+      }
+      await this.asyncSleep(1000);
+    }
+  }
+
+  /*
+   * Additional note:
+   * Only accept lock and type parameters as `Script` type, along with `data` field in QueryOptions. Use it carefully!
+   * */
+  collector(queries: CkbQueryOptions): CellCollector {
+    return new IndexerCollector(this, queries);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+  private async request(
+    method: string,
+    params?: any,
+    ckbIndexerUrl: string = this.ckbIndexerUrl
+  ): Promise<any> {
+    const data = {
+      id: 0,
+      jsonrpc: "2.0",
+      method,
+      params,
+    };
+    const res: rpcResponse = await axios.post(ckbIndexerUrl, data);
+    if (res.status !== 200) {
+      throw new Error(`indexer request failed with HTTP code ${res.status}`);
+    }
+    if (res.data.error !== undefined) {
+      throw new Error(
+        `indexer request rpc failed with error: ${JSON.stringify(
+          res.data.error
+        )}`
+      );
+    }
+    return res.data.result;
+  }
+
+  public async getCells(
+    searchKey: SearchKey,
+    terminator: Terminator = DefaultTerminator,
+    {
+      sizeLimit = 0x100,
+      order = Order.asc,
+      lastCursor = undefined,
+    }: AdditionalOptions = {}
+  ): Promise<GetCellsResults> {
+    const infos: Cell[] = [];
+    let cursor: string | undefined = lastCursor;
+    const index = 0;
+    while (true) {
+      let params = [searchKey, order, `0x${sizeLimit.toString(16)}`, cursor];
+      const res: GetLiveCellsResult = await this.request("get_cells", params);
+      const liveCells = res.objects;
+      cursor = res.last_cursor;
+      for (const liveCell of liveCells) {
+        const cell: Cell = {
+          cell_output: liveCell.output,
+          data: liveCell.output_data,
+          out_point: liveCell.out_point,
+          block_number: liveCell.block_number,
+        };
+        const { stop, push } = terminator(index, cell);
+        if (push) {
+          infos.push(cell);
+        }
+        if (stop) {
+          return {
+            objects: infos,
+            lastCursor: cursor,
+          };
+        }
+      }
+      if (liveCells.length < sizeLimit) {
+        break;
+      }
+    }
+    return {
+      objects: infos,
+      lastCursor: cursor,
+    };
+  }
+
+  running(): boolean {
+    return true;
+  }
+
+  start(): void {
+    debug.log("ckb indexer start");
+  }
+
+  startForever(): void {
+    debug.log("ckb indexer startForever");
```

Comment:
```suggestion
    debug.warn("deprecated: no need to startForever the ckb-indexer manually");
```

Reply:
follow lumos style and use DefaultLogger